src/java_bytecode/java_object_factory.cpp

```diff
@@ -56,38 +181,79 @@ void gen_nondet_init(
         // make null
         null_pointer_exprt null_pointer_expr(pointer_type);
         code_assignt code(expr, null_pointer_expr);
+        code.add_source_location()=loc;
         init_code.copy_to_operands(code);
 
         return;
       }
     }
 
-    // build size expression
-    exprt object_size=size_of_expr(subtype, ns);
-
-    if(subtype.id()!=ID_empty && !object_size.is_nil())
-    {
-      // malloc expression
-      side_effect_exprt malloc_expr(ID_malloc);
-      malloc_expr.copy_to_operands(object_size);
-      malloc_expr.type()=pointer_type;
+    code_labelt set_null_label;
+    code_labelt init_done_label;
 
-      code_assignt code(expr, malloc_expr);
-      init_code.copy_to_operands(code);
+    static size_t synthetic_constructor_count=0;
 
-      // dereference expression
-      dereference_exprt deref_expr(expr, subtype);
+    if(!assume_non_null)
+    {
+      auto returns_null_sym=
+          new_tmp_symbol(symbol_table, "opaque_returns_null");
```

Comment:
fixed, but I did not get a warning here, got> cpplint.py java_bytecode/java_object_factory.cpp 
Done processing java_bytecode/java_object_factory.cpp
Total errors found: 0
```
```